### PR TITLE
tso: fix a global TSO generation bug

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Fix a bug that will cause the Global TSO generation to be critically wrong. If `maxTSO.Logical` is bigger than `maxLogical`, it will be set to zero but without adding the count.

### What is changed and how it works?

Add the count to `maxTSO.Logical` after being set to zero.

### Check List

Tests

- Unit test
- Integration test

### CODE REVIEW

**Review by Yisaer — COMMENTED** [review]
Rest LGTM, plz add a unit test

**Review by Yisaer — APPROVED** [review]
LGTM

File: tests/server/tso/global_tso_test.go [hunk 1]
```diff
@@ -498,6 +518,41 @@ func (s *testSynchronizedGlobalTSO) testGetTimestamp(ctx context.Context, c *C,
...
+	defer cluster.Destroy()
+	c.Assert(err, IsNil)
+
+	err = cluster.RunInitialServers()
+	c.Assert(err, IsNil)
+
+	waitAllLeaders(s.ctx, c, cluster, dcLocationConfig)
+
+	for _, dcLocation := range dcLocationConfig {
+		pdName := cluster.WaitAllocatorLeader(dcLocation)
```
Comment (rleungx) [inline tests/server/tso/global_tso_test.go:546]: Is it duplicated with `waitAllLeaders`?

Comment (JmPotato) [reply tests/server/tso/global_tso_test.go]: Updated.

File: tests/server/tso/global_tso_test.go [hunk 2]
```diff
@@ -326,10 +330,9 @@ func (s *testTimeFallBackSuite) testGetTimestamp(c *C, n uint32) *pdpb.Timestamp
 	resp, err := tsoClient.Recv()
 	c.Assert(err, IsNil)
 	c.Assert(resp.GetCount(), Equals, uint32(n))
-
 	res := resp.GetTimestamp()
-	c.Assert(tsoutil.CompareTimestamp(res, tsoutil.GenerateTimestamp(time.Now(), 0)), Equals, 1)
-
+	c.Assert(res.GetPhysical(), Greater, int64(0))
```
Comment (rleungx) [inline tests/server/tso/global_tso_test.go:334]: Why do we need to change it?

Comment (JmPotato) [reply tests/server/tso/global_tso_test.go:334]: The original code compares `res` with zero. In addition to this, the new codes will also check whether the logical part is bigger than `count`.

Comment (rleungx) [reply tests/server/tso/global_tso_test.go:334]: It seems physical is not zero before?

Comment (JmPotato) [reply tests/server/tso/global_tso_test.go]: Oops, my mistake. Fixed.

**Review by rleungx — APPROVED** [review]

**Review by HunDunDM — APPROVED** [review]

**Pull request merged**